### PR TITLE
Cmsc495 34

### PR DESCRIPTION
**Pull request by bladejogger**
first and lastname added to User model
first and lastname are editable
.gitignore set to ignore .class files anywhere, and the folder 'target' in the top directory (maven artifact)

still needed:
unit tests
api documentation

### CODE REVIEW

File: src/main/java/com/group1/taskmanagement/config/DatabaseSeed.java [hunk 1]
```diff
@@ -31,6 +31,8 @@ public static void signUpUsers(JdbcTemplate jdbcTemplate, WebClient webClient) {
         if(count1 == 0) {
             Map<String, Object> map = new HashMap<>();
             map.put("username", "admin");
+            map.put("firstname", "");
```
Comment (ConnorDBurge) [inline src/main/java/com/group1/taskmanagement/config/DatabaseSeed.java]: Can we just leave these null (not include them) if were going to make them empty. That way we can just null check on the front end instead checking the length.

**Comment by ConnorDBurge** [comment]
I can sign a new user up. But the lastName and firstName are not saved. They are still set as null. 

image

image

And there also not saved during update of the user

image

File: src/main/java/com/group1/taskmanagement/model/User.java [hunk 2]
```diff
@@ -23,6 +23,12 @@ public class User {
     @Column(nullable = false, unique = true)
     private String username;
 
+    @Column(nullable = true, unique = false)
+    private String firstname;
+
+    @Column(nullable = true, unique = false)
+    private String lastname;
```
Comment (ConnorDBurge) [inline src/main/java/com/group1/taskmanagement/model/User.java:30]: Can you camelCase these variable names everywhere? The rest of the app is using camelCase. That way we don't run into any casing errors on the front end from inconsistency. I just ran into this issue when signing a new user up.

Also, you don't need a @Column with the nullable and unique declaration. Those are the default values for @Column.

**Review by ConnorDBurge — CHANGES_REQUESTED** [review]

File: src/main/java/com/group1/taskmanagement/security/CustomUserDetails.java [hunk 3]
```diff
@@ -29,13 +33,21 @@ public Collection<? extends GrantedAuthority> getAuthorities() {
...
+        return username;
+    }
+
+    public String getFirstName() {
+        return firstname;
+    }
+
+    public String getLastName() {
+        return lastname;
     }
```
Comment (ConnorDBurge) [inline src/main/java/com/group1/taskmanagement/security/CustomUserDetails.java]: We don't need these here either. The CustomUserDetails class is only used by Spring Security. Spring Security only needs the username and password.

File: src/main/java/com/group1/taskmanagement/model/User.java (hunk 2 above)
Comment (ConnorDBurge) [reply src/main/java/com/group1/taskmanagement/model/User.java:30]: Actually, I guess its fine. No need to change it. We can just keep at as the same case as `username`. Username is one word and first name is usually two words.

**Review by ConnorDBurge — CHANGES_REQUESTED** [review]

File: src/main/java/com/group1/taskmanagement/config/DatabaseSeed.java (hunk 1 above)
Comment (bladejogger) [reply src/main/java/com/group1/taskmanagement/config/DatabaseSeed.java]: i deleted them and it recompiled just fine; so, yes, we can!

File: src/main/java/com/group1/taskmanagement/security/CustomUserDetails.java (hunk 3 above)
Comment (bladejogger) [reply src/main/java/com/group1/taskmanagement/security/CustomUserDetails.java]: deleted it with no issues.

File: src/main/java/com/group1/taskmanagement/config/DatabaseSeed.java (hunk 1 above)
Comment (ConnorDBurge) [reply src/main/java/com/group1/taskmanagement/config/DatabaseSeed.java]: Awesome

File: src/main/java/com/group1/taskmanagement/security/CustomUserDetails.java (hunk 3 above)
Comment (ConnorDBurge) [reply src/main/java/com/group1/taskmanagement/security/CustomUserDetails.java]: Awesome

**Review by ConnorDBurge — APPROVED** [review]

**Comment by ConnorDBurge** [comment]
I did a deploy of this branch to Heroku. Everything works great.

**Comment by ConnorDBurge** [comment]
I'm going to merge it so I can have these changes on the ticket I'm starting

**Pull request merged**